core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java

```diff
@@ -449,6 +451,9 @@ private void runInternal() {
             });
           }
         } catch (Throwable e) {
+          if (isFatalError(e)) {
+            ProcessUtils.fatalError(LOG, e, "Error while reading");
```

Comment:
If you just want to output the error, maybe you cannot call `ProcessUtils.fatalError`, sometime, OutOfMemoryError can be cured by gc. 

```java
  public static void fatalError(Logger logger, Throwable t, String format, Object... args) {
    String message = String.format("Fatal error: " + format, args);
    if (t != null) {
      message += "\n" + Throwables.getStackTraceAsString(t);
    }
    if (Configuration.getBoolean(PropertyKey.TEST_MODE)) {
      throw new RuntimeException(message);
    }
    logger.error(message);
    System.exit(-1);
  }
```

Reply:
But it's kind of hard to tell whether it's OK to OOM or not. And it's a bit unpredictable what would happen when workers keep busy for a long time(we've seen clients hang or other weird behavior when workers OOM). We prefer to decommission the worker if he is in an unhealthy state(even though later on cured by gc) and let Devops decide what to do with that.

Reply:
If we just need to crash or exit the JVM, just add the jvm argument `-XX:+CrashOnOutOfMemoryError` or `-XX:+ExitOnOutOfMemoryError` , these can help user chose exit or not, but if we add a exit in alluxio code, user cannot   chose to keep process live.

In our cluster, we see the fuse process and worker process OOM because a huge amount of requests, but it can be healthy after those requests gone.

Reply:
I see. Let us discuss more internally and get back to you.

Reply:
@jja725 Is there any discussion or new feedback from internal?

Reply:
We still need some time to discuss this issue. Sorry for the lateness since it's the holiday season in US.

Reply:
@jja725 Thanks for the reply, it doesn't matter, hope to get the feedback.